File: data/snippets/quiz010.cpp
class EmailNotifier {
public:
    void sendNotification(const std::string& recipient, const std::string& subject, 
                        const std::string& message) {
        std::cout << "Sending email to: " << recipient << std::endl;
        std::cout << "Subject: " << subject << std::endl;
        std::cout << "Message: " << message << std::endl;
        std::cout << "Email sent successfully!" << std::endl;
    }
    
    bool checkDelivery(const std::string& messageId) {
        // 检查电子邮件是否已送达
        std::cout << "Checking email delivery status..." << std::endl;
        return true;
    }
};

class SMSAlertSystem {
public:
    // 功能类似但接口不同
    void alertUser(const std::string& phoneNumber, const std::string& alertContent) {
        std::cout << "Sending SMS alert to: " << phoneNumber << std::endl;
        std::cout << "Alert content: " << alertContent << std::endl;
        std::cout << "SMS alert sent successfully!" << std::endl;
    }
    
    bool hasBeenDelivered(const std::string& alertId) {
        // 检查短信是否已送达
        std::cout << "Checking SMS delivery status..." << std::endl;
        return true;
    }
};

class PushNotificationManager {
public:
    // 又一个类似功能但不同接口的类
    void pushAlert(const std::string& deviceToken, const std::string& title, 
                 const std::string& body, bool highPriority) {
        std::cout << "Pushing notification to device: " << deviceToken << std::endl;
        std::cout << "Title: " << title << std::endl;
        std::cout << "Body: " << body << std::endl;
        std::cout << "Priority: " << (highPriority ? "High" : "Normal") << std::endl;
        std::cout << "Push notification sent!" << std::endl;
    }
    
    bool verifyDelivery(const std::string& notificationId) {
        // 检查推送通知是否已送达
        std::cout << "Checking push notification delivery status..." << std::endl;
        return true;
    }
};

File: data/snippets/quiz010.java
class EmailNotifier {
    public void sendNotification(String recipient, String subject, String message) {
        System.out.println("Sending email to: " + recipient);
        System.out.println("Subject: " + subject);
        System.out.println("Message: " + message);
        System.out.println("Email sent successfully!");
    }
    
    public boolean checkDelivery(String messageId) {
        System.out.println("Checking email delivery status...");
        return true;
    }
    
    public void setEmailConfiguration(String smtpServer, int port, String username) {
        System.out.println("Email configuration set: " + smtpServer + ":" + port);
    }
}

class SMSAlertSystem {
    public void alertUser(String phoneNumber, String alertContent) {
        System.out.println("Sending SMS alert to: " + phoneNumber);
        System.out.println("Alert content: " + alertContent);
        System.out.println("SMS alert sent successfully!");
    }
    
    public boolean hasBeenDelivered(String alertId) {
        System.out.println("Checking SMS delivery status...");
        return true;
    }
    
    public void configureSMSProvider(String providerName, String apiKey) {
        System.out.println("SMS provider configured: " + providerName);
    }
}

class PushNotificationManager {
    public void pushAlert(String deviceToken, String title, String body, boolean highPriority) {
        System.out.println("Pushing notification to device: " + deviceToken);
        System.out.println("Title: " + title);
        System.out.println("Body: " + body);
        System.out.println("Priority: " + (highPriority ? "High" : "Normal"));
        System.out.println("Push notification sent!");
    }
    
    public boolean verifyDelivery(String notificationId) {
        System.out.println("Checking push notification delivery status...");
        return true;
    }
    
    public void setupPushService(String fcmServerKey, String apnsKeyPath) {
        System.out.println("Push service configured with FCM and APNS keys");
    }
}

class SlackMessageSender {
    public void postMessage(String channel, String messageText, String username) {
        System.out.println("Posting message to Slack channel: " + channel);
        System.out.println("Message: " + messageText);
        System.out.println("From: " + username);
        System.out.println("Slack message posted!");
    }
    
    public boolean isMessageSent(String messageTimestamp) {
        System.out.println("Checking Slack message status...");
        return true;
    }
    
    public void initializeSlackBot(String botToken, String workspaceId) {
        System.out.println("Slack bot initialized for workspace: " + workspaceId);
    }
}
